functions/consumer/geode-consumer/src/test/java/org/springframework/cloud/fn/consumer/geode/GeodeConsumerApplicationTests.java

```diff
@@ -0,0 +1,115 @@
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.apache.geode.cache.Region;
+import org.apache.geode.pdx.PdxInstance;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.cloud.fn.test.support.geode.GeodeContainer;
+import org.springframework.cloud.fn.test.support.geode.GeodeContainerIntializer;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.support.GenericMessage;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class GeodeConsumerApplicationTests {
+
+	private ApplicationContextRunner applicationContextRunner;
+
+	private GeodeContainer geode;
+
+	private ObjectMapper objectMapper = new ObjectMapper();
+
+	@BeforeAll
+	void setup() {
+		GeodeContainerIntializer initializer = new GeodeContainerIntializer(
+				geodeContainer -> {
+					geodeContainer.connectAndExecGfsh("create region --name=Stocks --type=REPLICATE");
+				});
+
+		applicationContextRunner = new ApplicationContextRunner()
+				.withUserConfiguration(GeodeConsumerTestApplication.class);
+
+		geode = initializer.geodeContainer();
+	}
+
+	@Test
+	void consumeWithJsonEnabled() {
+		applicationContextRunner
+				.withPropertyValues(
+						"geode.region.regionName=Stocks",
+						"geode.consumer.json=true",
+						"geode.consumer.key-expression=payload.getField('symbol')",
+						"geode.pool.hostAddresses=" + "localhost:" + geode.getLocatorPort())
+				.run(context -> {
+					Consumer<Message<?>> geodeConsumer = context.getBean("geodeConsumer", Consumer.class);
+
+					String json = objectMapper.writeValueAsString(new Stock("XXX", 100.00));
+					geodeConsumer.accept(new GenericMessage<>(json));
+
+					Region<String, PdxInstance> region = context.getBean(Region.class);
+					PdxInstance instance = region.get("XXX");
+					assertThat(instance.getField("price")).isEqualTo(100.00);
+					region.close();
+				});
+	}
+
+	@Test
+	void consumeWithoutJsonEnabled() {
+		applicationContextRunner
+				.withPropertyValues(
+						"geode.region.regionName=Stocks",
+						"geode.consumer.key-expression='key'",
+						"geode.pool.hostAddresses=" + "localhost:" + geode.getLocatorPort())
+				.run(context -> {
+					Consumer<Message<?>> geodeConsumer = context.getBean("geodeConsumer", Consumer.class);
+
+					geodeConsumer.accept(new GenericMessage<>("value"));
+
+					Region<String, String> region = context.getBean(Region.class);
+					String value = region.get("key");
+					assertThat(value).isEqualTo("value");
+					region.close();
+				});
+	}
+
+	@Data
+	@AllArgsConstructor
+	@NoArgsConstructor
+	static class Stock {
+		private String symbol;
+
+		private double price;
+	}
```

Comment:
Wonder if we should use Lombok here? I don't think we use Lombok anywhere else in the repo.

Reply:
Just for testing, why not?